topics/integrate-in-existing-app.md

```diff
@@ -1,242 +1,423 @@
-mainly contains platform-specific code, because it would be easier to maintain two specific versions of the module, one for Android and one for iOS.
+1. In Android Studio, click **File** | **New** | **New Module**.
 
-For example, a module that reads or writes from the device storage contains a lot of platform code like APIs to access files. 
-So it is better if each platform project maintains its own version of the module by implementing the same interface.
+2. In the list of templates, select **KMM Shared Module**, enter the module name `shared` and select the **Generate packForXcode Gradle task** checkbox.  
+    This task is required for connecting the shared module to the iOS application.
 
-On the other hand, a module that manages credentials may contain some platform-specific code (such as for encryption), 
-but it mostly provides the logic that is common for both platforms. That's why it's a perfect candidate for sharing.
+   ![KMM shared module](kmm-module-wizard-1.png)
 
-You can also choose to share the UI behavior using the [Model-View-Presenter (_MVP_)](#mvp-for-legacy-ui-frameworks) 
-or the [Model-View-Intent (_MVI_)](#mvi-for-declarative-ui-frameworks) pattern. These patterns:
+3. Click **Finish**.
 
-*   Make a clear distinction between the UI and presentation layers.
-*   Are completely decoupled from the UI platform.
-*   Are easily testable without a UI environment.
+The wizard creates the KMM shared module, updates configuration files, and creates files with classes that demonstrate multiplatform benefits.
+You can learn more about the [KMM project structure](discover-kmm-project.md).
+    
+### Depend on the shared module in your Android application
+
+To use cross-platform code in your Android application, connect the shared module to it, move the business logic code there, and make this code cross platform.
+
+1. Ensure that `compileSdkVersion` and `minSdkVersion` in `build.gradle.kts` of the `shared` module are the same as `build.gradle` of your Android application in the `app` module.  
+   If they are different, update them in `build.gradle.kts` of the shared module. Otherwise, you'll encounter a compile error.
+   
+2. Add a dependency on the shared module to `build.gradle` of your Android application.
+   
+    ```kotlin
+    dependencies {
+        implementation project(':shared')
+    }
+    ```
 
-#### MVP for legacy UI frameworks {initial-collapse-state="collapsed"}
+3. Apply the changes by synchronizing the project. 
 
-_Model-View-Presenter_ (_MVP_) forces you to create an API for both the Presenter that receives inputs and the View that displays outputs, allowing 
-you to test each independently.
+    ![Synchronize the Gradle files](gradle-sync.png)
 
-Here is an example of a simple MVP presenter:
+4. To check that the shared module is successfully connected to your application, dump the `greeting()` function result to the log 
```

Comment:
If this step is necessary, we can create a separate section to check, that everything works ok.